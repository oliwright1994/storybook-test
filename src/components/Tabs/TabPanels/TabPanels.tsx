import { TabPanels as ChakraTabPanels, TabPanelsProps as IChakraTabPanels } from '@chakra-ui/core'
import React, { FC } from 'react'

const TabPanels: FC<IChakraTabPanels> = ({ children }) => {
  return <ChakraTabPanels>{children}</ChakraTabPanels>
}

export default TabPanels
