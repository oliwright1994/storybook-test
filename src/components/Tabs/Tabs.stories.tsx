import createStory from '@bright/storybook/createStory'
import React from 'react'

import Tabs from '.'

createStory('COMPONENTS | Tabs', [
  {
    label: 'Tabs',
    jsx: () => {
      return (
        <Tabs>
          <Tabs.TabList>
            <Tabs.Tab>Tab1</Tabs.Tab>
            <Tabs.Tab>Tab2</Tabs.Tab>
            <Tabs.Tab>Tab2</Tabs.Tab>
          </Tabs.TabList>
          <Tabs.TabPanels>
            <Tabs.TabPanel>Panel One</Tabs.TabPanel>
            <Tabs.TabPanel>Panel Two</Tabs.TabPanel>
            <Tabs.TabPanel>Panel Three</Tabs.TabPanel>
          </Tabs.TabPanels>
        </Tabs>
      )
    },
  },
])
