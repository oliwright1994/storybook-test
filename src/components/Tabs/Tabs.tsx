import { ITabs as IChakraTabs, Tabs as ChakraTabs } from '@chakra-ui/core'
import React, { FC } from 'react'

import Tab from './Tab'
import TabList from './TabList'
import TabPanel from './TabPanel'
import TabPanels from './TabPanels'

interface ITabs extends FC<IChakraTabs> {
  Tab: typeof Tab
  TabList: typeof TabList
  TabPanel: typeof TabPanel
  TabPanels: typeof TabPanels
}

const Tabs: ITabs = ({ children }) => {
  return <ChakraTabs  flexGrow={1}>{children}</ChakraTabs>
}

Tabs.Tab = Tab
Tabs.TabList = TabList
Tabs.TabPanel = TabPanel
Tabs.TabPanels = TabPanels

export default Tabs
