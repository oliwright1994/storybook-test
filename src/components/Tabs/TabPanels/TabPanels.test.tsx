import React from 'react'
import { renderWithBrightTheme } from 'testing/helpers/themeHelper'

import Tabs from '../Tabs'
import TabPanels from '.'

describe('<TabPanels />', () => {
  const getLabel = (id: string): string => `Panel${id}`
  const panels = ['1', '2'].map(id => <Tabs.TabPanel key={id}>{getLabel(id)}</Tabs.TabPanel>)

  it('should render as expected', () => {
    const { container } = renderWithBrightTheme(
      <Tabs>
        <TabPanels>{panels}</TabPanels>
      </Tabs>
    )

    expect(container).toMatchSnapshot()
  })

  it('should render child panels as expected', () => {
    const { getByText } = renderWithBrightTheme(
      <Tabs>
        <TabPanels>{panels}</TabPanels>
      </Tabs>
    )

    expect(getByText(getLabel('1'))).toBeTruthy()
    expect(getByText(getLabel('2'))).toBeTruthy()
  })
})
