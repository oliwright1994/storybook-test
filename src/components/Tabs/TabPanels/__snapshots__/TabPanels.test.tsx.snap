// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabPanels /> should render as expected 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      class="css-0"
      tabindex="-1"
    >
      <div
        aria-labelledby="tab:undefined"
        class="css-1jj07fx"
        id="panel:undefined"
        role="tabpanel"
        tabindex="-1"
      >
        Panel1
      </div>
      <div
        aria-labelledby="tab:undefined"
        class="css-1jj07fx"
        hidden=""
        id="panel:undefined"
        role="tabpanel"
        tabindex="-1"
      >
        Panel2
      </div>
    </div>
  </div>
</div>
`;
